assets/src/edit-story/components/helpCenter/useHelpCenter/index.js

```diff
@@ -167,12 +163,23 @@ export function useHelpCenter() {
 
   const persistenceKey = createKeyFromBooleanMap(store.state.readTips);
   useEffect(() => {
-    persistenceKey &&
-      updateCurrentUser({
-        meta: {
-          web_stories_onboarding: createBooleanMapFromKey(persistenceKey),
-        },
-      }).catch(actions.persistingReadTipsError);
+    if (!persistenceKey) {
+      return () => {};
+    }
+    // The call to `updateCurrentUser` causes the entire app to rerender
+    // which causes noticable jank in our transition. This is just
```

Comment:
Open to suggestions on how we could avoid this re-render :-)

Reply:
Could make the `useCurrentUser` use a store instead of a provider to make its state global. When the `currentUser` updates only nodes subscribed to the store and their children will get diffed by react.

[zustand](https://www.npmjs.com/package/zustand) or [react-hooks-global-state](https://github.com/dai-shi/react-hooks-global-state) come to mind. I think I slightly prefer zustand just because of wider adoption and the api, but open to either. both are great implementations and super small bundle size

Reply:
With `use-context-selector` (which we use) this would also be possible, but only when also using `React.memo`, if I am not mistaken?

Reply:
Yea I thought so too. Maybe there's a child consuming the context high up in the app hierarchy? which is making it seem like an app wide render when it's actually just a big portion of the app re-rendering.

Worth creating a ticket to look into this so we don't have to dance around it in this component. @BrittanyIRL you agree?

Reply:
i think a new ticket makes sense that way it can be investigated outside of holding up the release potentially.

Reply:
Agreed. the fix doesn't really have anything to do with the code we introduced. it just highlights it. Want me to create a ticket real quick?

Reply:
https://github.com/google/web-stories-wp/issues/6309

Reply:
Thanks!